api/src/main/java/org/ehcache/spi/persistence/StateRepository.java

```diff
@@ -33,7 +52,8 @@
    * @param valueClass concrete value type
    * @param <K> the key type, must be {@code Serializable}
    * @param <V> the value type, must be {@code Serializable}
+   * @param whiteListedClasses classes whose objects would get deserialized as part of key or value deserialization
    * @return a state holder
    */
-  <K extends Serializable, V extends Serializable> StateHolder<K, V> getPersistentStateHolder(String name, Class<K> keyClass, Class<V> valueClass);
+  <K extends Serializable, V extends Serializable> StateHolder<K, V> getPersistentStateHolder(String name, Class<K> keyClass, Class<V> valueClass, Iterable<Class> whiteListedClasses);
```

Comment:
This is a breaking API change, not sure what policy is on this atm.  Would it be better or worse to have the default implementation call the original non-whitelisting method?

Reply:
Because this is API, I would much prefer we deprecate the existing method and add a new one. Also I do not see the benefit of having both the vararg and `Iterable` versions.

Reply:
retained the original but marked as deprecated, added new one with Collection<Class<?>> as extra parameter with default implementation which ignores the passed in whitelist.

Reply:
It seems wrong to have the default implementation not honor the contract properly.  Shouldn't it be the non-whitelisting implementation that has a default implementation that calls the whitelisting one with a "infinite whitelist"?

Reply:
Added a default implementation for non-whitelisted one now that calls whitelisted one with predicate as c->true and null classLoader.